code/game/machinery/recycler.dm

```diff
@@ -64,8 +64,21 @@
 		return TRUE
 
 /obj/machinery/recycler/screwdriver_act(mob/user, obj/item/I)
-	if(default_deconstruction_screwdriver(user, "grinder-oOpen", "grinder-o0", I))
-		return TRUE
+	. = TRUE
+	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+		return
+	if(panel_open)
+		panel_open = FALSE
+	else
+		panel_open = TRUE
+	update_icon(UPDATE_OVERLAYS)
+
+/obj/machinery/recycler/update_overlays()
+	. = ..()
+	if(panel_open == TRUE)
+		. += "grinder-oOpen"
+	else
+		. -= "grinder-oOpen"
```

Comment:
It's just an overlay that is being added/removed depending on the panel state (either += or -=)

Reply:
The parent `update_overlays` always returns an empty list that gets filled with overlays to be added. The `. -= "grinder-oOpen"` is redundant because it always gets removed when the proc is called.

Reply:
That's good to know, thanks! I will remove that part then.